mp/src/game/client/neo/ui/neo_hud_friendly_marker.cpp

```diff
@@ -66,87 +69,103 @@ CNEOHud_FriendlyMarker::CNEOHud_FriendlyMarker(const char* pElemName, vgui::Pane
 
 	SetVisible(true);
 }
-
-void CNEOHud_FriendlyMarker::UpdateStateForNeoHudElementDraw()
+void CNEOHud_FriendlyMarker::DrawNeoHudElement()
 {
-	int x, y;
+	if (!ShouldDraw())
+	{
+		return;
+	}
 	const float scale = neo_friendly_marker_hud_scale_factor.GetFloat();
-	const float heightOffset = 48.0f;
-	Vector pos;
-
-	auto localPlayer = C_NEO_Player::GetLocalNEOPlayer();
-
-	Assert(localPlayer->m_rvFriendlyPlayerPositions.Count() == MAX_PLAYERS);
 
-	COMPILE_TIME_ASSERT(ARRAYSIZE(m_x0) == MAX_PLAYERS);
-	COMPILE_TIME_ASSERT(ARRAYSIZE(m_x1) == MAX_PLAYERS);
-	COMPILE_TIME_ASSERT(ARRAYSIZE(m_y0) == MAX_PLAYERS);
-	COMPILE_TIME_ASSERT(ARRAYSIZE(m_y1) == MAX_PLAYERS);
+	m_iMarkerWidth = (m_iMarkerTexWidth * 0.5f) * scale;
+	m_iMarkerHeight = (m_iMarkerTexHeight * 0.5f) * scale;	
 
-	for (int i = 0; i < MAX_PLAYERS; ++i)
+	auto localPlayer = C_NEO_Player::GetLocalNEOPlayer();
+	auto team = localPlayer->GetTeam();
+	m_IsSpectator = team->GetTeamNumber() == TEAM_SPECTATOR;
+	auto memberCount = team->GetNumPlayers();
+	
+	if(m_IsSpectator)
 	{
-		if ((localPlayer->entindex() == i + 1) || // Skip self
-			(localPlayer->m_rvFriendlyPlayerPositions[i] == vec3_origin)) // Skip unused positions
+		for(int t = 0; t < 2; ++t)
 		{
-			continue;
+			team = GetGlobalTeam(TEAM_SPECTATOR + t + 1);
+			memberCount = team->GetNumPlayers();
+			for (int i = 0; i < memberCount; ++i)
+			{
+				auto player = team->GetPlayer(i);
+				if(player && player->IsAlive())
+				{
+					DrawPlayer(GetTeamColour(team->GetTeamNumber()), player);
+				}		
+			}
 		}
```

Comment:
Just loop over the clients and check the team per player, offsetting by spec is a bit odd.
Either that or put an assert to make sure team given are Jinrai or NSF.